src/common/meta/src/kv_backend/postgres.rs

```diff
@@ -75,77 +77,184 @@ impl PgQueryExecutor<'_> {
+    }
+
+    /// Generates the sql for batch delete.
+    fn generate_batch_delete_query(&self, key_len: usize) -> String {
+        let table_name = &self.table_name;
+        let in_clause = generate_in_placeholders(1, key_len).join(", ");
+        format!(
+            "DELETE FROM {table_name} WHERE k in ({}) RETURNING k,v;",
+            in_clause
+        )
+    }
+
+    /// Generates the sql for batch upsert.
+    fn generate_batch_upsert_query(&self, kv_len: usize) -> String {
+        let table_name = &self.table_name;
+        let in_placeholders: Vec<String> = (1..=kv_len).map(|i| format!("${}", i)).collect();
+        let in_clause = in_placeholders.join(", ");
+        let mut param_index = kv_len + 1;
+        let mut values_placeholders = Vec::new();
+        for _ in 0..kv_len {
+            values_placeholders.push(format!("(${0}, ${1})", param_index, param_index + 1));
+            param_index += 2;
+        }
+        let values_clause = values_placeholders.join(", ");
 
-const PREFIX_DELETE: &str = "DELETE FROM greptime_metakv WHERE k LIKE $1 RETURNING k,v;";
+        format!(
+            r#"
+    WITH prev AS (
+        SELECT k,v FROM {table_name} WHERE k IN ({in_clause})
+    ), update AS (
+    INSERT INTO {table_name} (k, v) VALUES
+        {values_clause}
+    ON CONFLICT (
+        k
+    ) DO UPDATE SET
+        v = excluded.v
+    )
 
-const RANGE_DELETE_LEFT_BOUNDED: &str = "DELETE FROM greptime_metakv WHERE k >= $1 RETURNING k,v;";
+    SELECT k, v FROM prev;
+    "#
+        )
+    }
+}
 
-const RANGE_DELETE_FULL_RANGE: &str =
-    "DELETE FROM greptime_metakv WHERE k >= $1 AND K < $2 RETURNING k,v;";
+/// Default sql template set for [`KeyValue`].
+pub type DefaultSqlTemplateSet = SqlTemplateSet<KeyValue>;
 
-const CAS: &str = r#"
-WITH prev AS (
-    SELECT k,v FROM greptime_metakv WHERE k = $1 AND v = $2
-), update AS (
-UPDATE greptime_metakv
-SET k=$1,
-v=$2
-WHERE 
-    k=$1 AND v=$3
-)
+/// Default pg store for [`KeyValue`].
+pub type DefaultPgStore = PgStore<KeyValue>;
 
-SELECT k, v FROM prev;
-"#;
+impl<T> PgStore<T> {
+    async fn client(&self) -> Result<PgClient> {
+        match self.pool.get().await {
+            Ok(client) => Ok(client),
+            Err(e) => GetPostgresConnectionSnafu {
+                reason: e.to_string(),
+            }
+            .fail(),
+        }
+    }
 
-const PUT_IF_NOT_EXISTS: &str = r#"    
-WITH prev AS (
-    select k,v from greptime_metakv where k = $1
-), insert AS (
-    INSERT INTO greptime_metakv
-    VALUES ($1, $2)
-    ON CONFLICT (k) DO NOTHING
-)
+    async fn client_executor(&self) -> Result<PgQueryExecutor<'_>> {
+        let client = self.client().await?;
+        Ok(PgQueryExecutor::Client(client))
+    }
 
-SELECT k, v FROM prev;"#;
+    async fn txn_executor<'a>(&self, client: &'a mut PgClient) -> Result<PgQueryExecutor<'a>> {
+        let txn = client
+            .build_transaction()
+            .isolation_level(IsolationLevel::Serializable)
```

Comment:
The default isolation level is？ Do we need to bench again?

Reply:
new bench here.
![image](https://github.com/user-attachments/assets/f1667d21-bc39-45c0-b075-4ce2d406436b)